rtengine/diagonalcurves.cc

```diff
@@ -434,21 +433,22 @@ double DiagonalCurve::getVal (double t) const
         if (t <= 1e-14) {
             return 0.0;
         }
-
-        double tv = xexp(mc * xlog(t));
+    
+        double tv = pow(t, mc);
         double base = pfull (tv, x[8], x[6], x[5]);
-        double stretched = base <= 1e-14 ? 0.0 : xexp(xlog(base) / mc);
-
+        double stretched = pow(base, 1 / mc);
```

Comment:
For clarity, please write `1.0`.
As far as I see, `mc` has the approximate range of 0.1 .. 69.0 (inverted between 0.0145 .. 9.966). The value of `base` can be between 0..1. So `stretched` can easily get very, very small. I'm not sure removing the guard is such a good idea...

Reply:
Basically for reasonable parameter values `base` and other variable operates in these small range of values, and clamping them produces artefacts in the curve. Maybe there's way to rework how the curve is computed to avoid that but I don't understand it enough to do it myself. That said I don't think there's really a fundamental issue with using these small values, as long as it doesn't produces infinities and NaNs.